Assets/_BForBoss/_Weapons/Scripts/ScriptableObjects/Melee/MeleeScriptableObject.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Perigon.Weapons
+{
+    public interface IMeleeProperties
+    {
+        float Damage { get; }
+        float AttackCoolDown { get; }
+
+        int OverlapCube(Vector3 position, Quaternion rotation, Collider[] buffer);
+    }
+    
+    [CreateAssetMenu(fileName = "MeleeProperties", menuName = "PerigonGames/Melee", order = 1)]
+    public class MeleeScriptableObject : ScriptableObject, IMeleeProperties
+    {
+        [SerializeField] private float _damage = 5f;
+        [SerializeField] private float _coolDown = 0.5f;
+        
+        [field: SerializeField] public float Range { get; set; } = 1f;
+        [field: SerializeField] public float Height { get; set; } = 1f;
+        [field: SerializeField] public float Width { get; set; } = 1f;
+        
+        public float Damage => _damage;
+        public float AttackCoolDown => _coolDown;
+
+        public Vector3 HalfExtents => new Vector3(Width, Height, Range) * 0.5f;
+
+        public int OverlapCube(Vector3 position, Quaternion rotation, Collider[] buffer)
```

Comment:
what does this do?  and why is it returning an int?

Reply:
Its basically just a proxy for Physics.OverlapCube. Returns and everything are the same

Reply:
okay i think i understand, is there a reason why it's in the scriptable object as oposed to the Melee script?

Reply:
i'm guessing it's related to the not having unity code within the melee.cs code?

Reply:
Is it possible to make the buffer `out buffer` ? so anyone else using this knows that it's the buffer that get's updated.

Reply:
Not so much not having unity code, but more that I didn't really think Melee should have access to the bounding box parameters. I can't make it an out parameter, but ref should work and that would be more clear

Reply:
@m4rtint just saw [BFB-287](https://perigongames.atlassian.net/browse/BFB-287), should I change the cube for a capsule?